dev/system-test/firestore.ts

```diff
@@ -3844,6 +3625,236 @@ describe.skip('Aggregation queries', () => {
+            year: 1980,
+            rating: 5,
+          },
+          b: {
+            author: 'authorB',
+            title: 'titleB',
+            pages: 50,
+            year: 2020,
+            rating: 4,
+          },
+          c: {
+            author: 'authorC',
+            title: 'titleC',
+            pages: 100,
+            year: 1980,
+            rating: 3,
+          },
+          d: {
+            author: 'authorD',
+            title: 'titleD',
+            pages: 50,
+            year: 2020,
+            rating: 0,
+          },
+        };
+        await addTestDocs(testDocs);
+        const snapshot = await col
+          .where('rating', 'in', [5, 3])
+          .aggregate({
+            totalRating: AggregateField.sum('rating'),
+            averageRating: AggregateField.average('rating'),
+            totalPages: AggregateField.sum('pages'),
+            averagePages: AggregateField.average('pages'),
+            countOfDocs: AggregateField.count(),
+          })
+          .get();
+        expect(snapshot.data().totalRating).to.equal(8);
+        expect(snapshot.data().averageRating).to.equal(4);
+        expect(snapshot.data().totalPages).to.equal(200);
+        expect(snapshot.data().averagePages).to.equal(100);
+        expect(snapshot.data().countOfDocs).to.equal(2);
+      });
+
+      it('performs aggregates when using `array-contains-any` operator', async () => {
+        const testDocs = {
+          a: {
+            author: 'authorA',
+            title: 'titleA',
+            pages: 100,
+            year: 1980,
+            rating: [5, 1000],
+          },
+          b: {
+            author: 'authorB',
+            title: 'titleB',
+            pages: 50,
+            year: 2020,
+            rating: [4],
+          },
+          c: {
+            author: 'authorC',
+            title: 'titleC',
+            pages: 100,
+            year: 1980,
+            rating: [2222, 3],
+          },
+          d: {
+            author: 'authorD',
+            title: 'titleD',
+            pages: 50,
+            year: 2020,
+            rating: [0],
+          },
+        };
+        await addTestDocs(testDocs);
+        const snapshot = await col
+          .where('rating', 'array-contains-any', [5, 3])
+          .aggregate({
+            totalRating: AggregateField.sum('rating'),
+            averageRating: AggregateField.average('rating'),
+            totalPages: AggregateField.sum('pages'),
+            averagePages: AggregateField.average('pages'),
+            countOfDocs: AggregateField.count(),
+          })
+          .get();
+        expect(snapshot.data().totalRating).to.equal(0);
+        expect(snapshot.data().averageRating).to.be.null;
+        expect(snapshot.data().totalPages).to.equal(200);
+        expect(snapshot.data().averagePages).to.equal(100);
+        expect(snapshot.data().countOfDocs).to.equal(2);
+      });
+
+      it('performs aggregations on nested map values', async () => {
```

Comment:
This test can be modified to run against production by dropping the aggregations over the second nested field `'metadata.rating.critic'`. For your consideration.

Reply:
Done